### PR TITLE
arch: Support aws graviton

### LINKED ISSUE #1106: Add ARM testing CI through github actions

### PR DESCRIPTION
**Pull request by mloubout**
AWS gravition arch and aws-ec2 on-demand self-hosted runners

Fixes #1106

### CODE REVIEW

File: devito/arch/compiler.py [hunk 1]
```diff
@@ -373,7 +373,7 @@ def add_ldflags(self, flags):
 class GNUCompiler(Compiler):
 
     def __init__(self, *args, **kwargs):
-        super(GNUCompiler, self).__init__(*args, **kwargs)
```
Comment (ggorman) [inline devito/arch/compiler.py:376]: I don't understand why this line has changed for graviton support.

File: devito/arch/compiler.py [hunk 2]
```diff
@@ -400,6 +400,17 @@ def __lookup_cmds__(self):
         self.MPICXX = 'mpicxx'
 
 
+class ArmCompiler(GNUCompiler):
```
Comment (ggorman) [inline devito/arch/compiler.py:403]: How is this different from previous ARM support?

Comment (mloubout) [reply devito/arch/compiler.py:403]: We did not have an ARM compiler before only gcc, this adds the extra flags for graviton and I'm sure other "brands" will need some flags too.

File: devito/arch/compiler.py (hunk 1 above)
Comment (mloubout) [reply devito/arch/compiler.py:376]: Oh it's just we moving towards `super()` everywhere in the codebase rather than with args so changed it while there

**Comment by ggorman** [comment]
Doesn't this fix https://github.com/devitocodes/devito/issues/1106

File: .github/workflows/cancel.yml [hunk 3]
```diff
@@ -12,5 +12,5 @@ jobs:
         with:
           # Ids to cancel core/mpi/gpu/examples/tutorials
           # https://api.github.com/repos/devitocodes/devito/actions/workflows
-          workflow_id: 167582, 203470, 203471, 434862, 501517
+          workflow_id: 167582, 203470, 203471, 434862, 501517, 50473775
```
Comment (ggorman) [inline .github/workflows/cancel.yml:15]: Ok - school me... what does this do?

Comment (mloubout) [reply .github/workflows/cancel.yml:15]: Oh this cancel the workflows currently running when you push again so we don't have thousands of concurrent runs from different commits on the same branch

Comment (mloubout) [reply .github/workflows/cancel.yml:15]: In the url line 14

File: devito/arch/compiler.py [hunk 4]
```diff
@@ -400,6 +400,17 @@ def __lookup_cmds__(self):
...
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        platform = kwargs.pop('platform', configuration['platform'])
+        # Graviton flag
+        if platform is GRAVITON:
+            self.cflags += ['-mcpu=neoverse-n1']
```
Comment (FabioLuporini) [inline devito/arch/compiler.py:411]: really =native doesn't do the same?

**Review by FabioLuporini — APPROVED** [review]
I left a small question, but otherwise GTG. Great, thanks

**Review by georgebisbas — APPROVED** [review]

File: devito/arch/compiler.py (hunk 4 above)
Comment (mloubout) [reply devito/arch/compiler.py:411]: According to the aws this flag should be used yeah

**Review by ggorman — APPROVED** [review]

**Pull request merged**